### ISSUE TITLE
usage examples

### ISSUE THREAD

**Issue by nwtn** [Webdevdata/webdevdata.org#3]
It would be awesome to see some typical examples of how to search through these data for specific tags, etc. It could eliminate a barrier to use.

**Comment by marcoscaceres** [comment]
@nwtn agreed. Do you want to volunteer to do them? They are just a few simple greps from the command line.

**Comment by nwtn** [comment]
sure

**Comment by marcoscaceres** [comment]
Here is a somewhat crappy one:

```
find ./ -print | xargs grep -l picturefill
```

**Comment by marcoscaceres** [comment]
Better one, thanks to @yoavweiss. Finds "apple-touch-icon"s in the HTML files, and spits out the count

```
find ./ -name "*ml.txt" |  xargs grep -l apple-touch-icon | wc -l
```

**Comment by oli** [comment]
If you want to use the included tools, check out:

### [Webdevdata/webdevdata-tools](https://github.com/Webdevdata/webdevdata-tools)

These tools produce comma-separated output, with one line per matched page.
- `wdd_select [-atrs=attr1,attr2...] [CSS selector] [file]`
- `wdd_meta_names [file]`
- `wdd_html_manifest [file]`
- `wdd_tag_count [file]`

### [baptistelebail/webdevdata.org](https://github.com/baptistelebail/webdevdata.org/)

These produce semi-colon-separated summaries. For `webdevdata-query.sh` this includes:
- CSS Query used
- Total number of instances
- Total number of pages with feature
- Max number of instances per page

[Refer to the Wiki page](https://github.com/baptistelebail/webdevdata.org/wiki/Queries-on-WebDevData) for details and examples
- `webdevdata-query.sh`
- `webdevdata-stats-HTML-attributes.sh`
- `webdevdata-stats-HTML-tags.sh`

`webdevdata-query.sh` took about 40 minutes per pass on the 2013-10 dataset for me, regardless of the number of CSS-like query terms, so if you’re querying multiple things (e.g. all the sectioning elements) list them all in one query, e.g.:

```
./webdevdata-query.sh webdevdata.org-2013-10-30-231036 body article section nav h1 h2 h3 h4 h5 h6 hgroup main
```

HTH

**Comment by marcoscaceres** [comment]
@oli, so I think what we are going to do is allow each repo provide examples of it's own usage. The front page of webdevdata.org is currently very poorly maintained :(

**Comment by oli** [comment]
@marcoscaceres here are some more to get you going then:
- Count the number of files containing `<html` (`-i`: case-insensitive, `-l`: stop on first match (faster), via `xargs` to grep content not filenames):

``` sh
find ./ -type f | xargs grep -il "<html" | wc -l
```
- Find files with the extension ".assembler" (`-name`), then execute `head` on each one (`'{}' +`) to display the first two lines (`-n 2`):

``` sh
find ./ -type f -name "*.assembler" -exec head -n 2 '{}' +
```
- Count the number of non-header files in the corpus that are less than 100 characters (`-size -100c`):

``` sh
find ./ -type f -not -name "*.hdr.txt" -size -100c | wc -l
```

HTH!

**Comment by marcoscaceres** [comment]
@oli super helpful! Thanks so much for all these! Ok, we now have a pretty good set to show how this all works. 

Will probably just start by collating all these and adding them to the README.
